core/src/main/scala/chisel3/internal/Error.scala

```diff
@@ -85,6 +85,14 @@ object ExceptionHelpers {
 
 }
 
+object InternalError {
```

Comment:
Did this a whole new way. Just made a extended excxeption, but it needs to be called with `throw new` so it would not break case statements that had to be a specific type